zkevm-circuits/src/util/word.rs

```diff
@@ -100,6 +117,90 @@ impl<F: Field, const N: usize> WordLimbs<Cell<F>, N> {
         })
     }
 
+    /// assign bytes to wordlimbs first half/second half respectively
+    // N_LO, N_HI are number of bytes to assign to first half and second half of size N limbs,
+    // respectively N_LO and N_HI can be different size, the only requirement is N_LO % (N/2)
+    // and N_HI % (N/2) [N/2] limbs will be assigned separately.
+    // E.g. N_LO = 4 => [nl1, nl2, nl3, nl4]
+    // N_HI = 2 => [nh1, nh2]
+    // N = 2 => [l1, l2]
+    // it equivalent l1.assign(nl1.expr() + nl2.expr() * 256 + nl3.expr() * 256^2 +  nl3.expr() *
+    // 256^3) and l2.assign(nh1.expr() + nh2.expr() * 256)
+    pub fn assign_lo_hi<const N_LO: usize, const N_HI: usize>(
+        &self,
+        region: &mut CachedRegion<'_, '_, F>,
+        offset: usize,
+        bytes_lo: Option<[u8; N_LO]>,
+        bytes_hi: Option<[u8; N_HI]>,
```

Comment:
My option is assign lo/hi separately still better than padding way since it do not need to waste extra space just for zero padding. Besides it pretty flexible. 

It just used by `assign_u256`, `assign_h160`, `assign_u64`, so I changed visibility to `private` in case of confusing.

Reply:
Sounds good to me.